pkg/cmd/kafka/topic/consume/consume.go

```diff
@@ -0,0 +1,244 @@
+
+	return cmd
+}
+
+func runCmd(opts *options) error {
+
+	conn, err := opts.f.Connection(connection.DefaultConfigRequireMasAuth)
+	if err != nil {
+		return err
+	}
+
+	api, _, err := conn.API().KafkaAdmin(opts.kafkaID)
+	if err != nil {
+		return err
+	}
+
+	if opts.wait {
+		err := consumeAndWait(opts, api)
+		if err != nil {
+			return err
+		}
+	} else {
+		records, err := consume(opts, api)
+		if err != nil {
+			return err
+		}
+
+		outputRecords(opts, records)
+	}
+
+	return nil
+}
+
+func consumeAndWait(opts *options, api *kafkainstanceclient.APIClient) error {
+	if opts.limit != DefaultLimit {
+		opts.f.Logger.Info(opts.f.Localizer.MustLocalize("kafka.topic.consume.log.info.limitIgnored", localize.NewEntry("Limit", DefaultLimit)))
+		opts.limit = DefaultLimit
+	}
+
+	if opts.offset != DefaultOffset {
+		opts.f.Logger.Info(opts.f.Localizer.MustLocalize("kafka.topic.consume.log.info.offsetIgnored", localize.NewEntry("Offset", DefaultOffset)))
+		opts.offset = DefaultOffset
+	}
+
+	if opts.from == DefaultTimestamp {
+		// get current time in ISO 8601
+		opts.from = time.Now().Format(time.RFC3339)
+	}
+
+	max_offset := opts.offset
+	first_consume := true
+	for true {
+
+		records, err := consume(opts, api)
+		if err != nil {
+			return err
+		}
+
+		record_count := len(records.Items)
+		if record_count > 0 {
+			max_offset = *(records.Items[record_count-1].Offset) + 1
+			outputRecords(opts, records)
+
+			if first_consume {
+				// reset timestamp after first consume as it will conflict with
+				// the max offset we are setting to only get new records
+				opts.from = DefaultTimestamp
+				first_consume = false
+			}
+		}
+
+		opts.offset = max_offset
+
+		time.Sleep(1 * time.Second)
+	}
+
+	return nil
+}
+
+func consume(opts *options, api *kafkainstanceclient.APIClient) (*kafkainstanceclient.RecordList, error) {
+
+	request := api.RecordsApi.ConsumeRecords(opts.f.Context, opts.topicName).Limit(opts.limit).Partition(opts.partition).Offset(int32(opts.offset))
+	if opts.from != DefaultTimestamp {
+
+		_, err := time.Parse(time.RFC3339, opts.from)
+		if err != nil {
+			return nil, errors.New(opts.f.Localizer.MustLocalize("kafka.topic.comman.error.timeFormat", localize.NewEntry("Time", opts.from)))
+		}
+
+		request = request.Timestamp(opts.from)
+	}
+
+	list, _, err := request.Execute()
+	if err != nil {
+		return nil, err
+	}
+
+	return &list, nil
+}
+
+func outputRecords(opts *options, records *kafkainstanceclient.RecordList) {
+	recordsAsRows := mapRecordsToRows(opts.topicName, &records.Items)
+
+	if len(records.Items) == 0 {
+		opts.f.Logger.Info(opts.f.Localizer.MustLocalize("kafka.common.log.info.noRecords"))
+		return
+	}
+
+	format := opts.outputFormat
+	if format == dump.EmptyFormat {
+		format = dump.JSONFormat
+	}
+
+	for i := 0; i < len(recordsAsRows); i++ {
+		row := &recordsAsRows[i]
+		if format == FormatKeyValue {
+			if row.Key == "" {
+				opts.f.Logger.Info(row.Value)
```

Comment:
We will miss message delimiter. Maybe `Value: ..` be better here to see when mesage finished.